<h1 align="middle"> Hospital Analytics </h1>
Hospital readmissions within short timeframes can signal chronic conditions, gaps in post-discharge care, or recurring treatment needs.<br> 
This project analyzes patient encounter and procedure data between 2011 - 2022 to identify patients readmitted within 30 days, highlight those with the most frequent readmissions, and explore the procedures associated with their care.<br>
Understanding these patterns is crucial for improving patient outcomes, optimizing hospital resources, and implementing targeted interventions that reduce avoidable short-term readmissions.

<h2 align="middle">1.0 Objectives</h2>

1. Encounters Overview
- How many total encounters occurred each year?
- For each year, what percentage of all encounters belonged to each encounter class (ambulatory, outpatient, wellness, urgent care, emergency, and inpatient)?
- What percentage of encounters were over 24 hours versus under 24 hours?
2. Cost & Coverage Insights
- How many encounters had zero payer coverage, and what percentage of total encounters does this represent?
- What are the top 10 most frequent procedures performed and the average base cost for each?
- What are the top 10 procedures with the highest average base cost and the number of times they were performed?
- What is the average total claim cost for encounters, broken down by payer?
3. Patient Behavior Analysis
- How many unique patients were admitted each quarter over time?
- How many patients were readmitted within 30 days of a previous encounter?
- Which patients had the most readmissions?

<h2 align="middle">2.0 Connecting to SQL</h2>

In [1]:
%load_ext sql

In [2]:
#!pip install ipython-sql
#!pip install ipython-sql pymysql
#!pip install pymysql

In [3]:
# Connecting to MySQL database
%sql mysql+pymysql://root:lilian80@localhost:3306/hospital_db

Connecting to 'mysql+pymysql://root:***@localhost:3306/hospital_db'

In [4]:
# Limiting the number of output to 100 for the large datasets
%config SqlMagic.displaylimit = 100

<h2 align="middle"> 3.0 Data Overview</h2>

In [5]:
%%sql
-- determining the tables in the database and the number of rowseach of them contains
SELECT 
    TABLE_NAME,
    TABLE_ROWS
FROM information_schema.tables
WHERE table_schema = 'hospital_db';

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

4 rows affected.

+------------+------------+
| TABLE_NAME | TABLE_ROWS |
+------------+------------+
| encounters |   27132    |
|  patients  |    973     |
|   payers   |     10     |
| procedures |   47280    |
+------------+------------+

In [6]:
%%sql
-- Patients table overview
SELECT *
FROM patients
LIMIT 3;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

3 rows affected.

+--------------------------------------+------------+-----------+--------+------------+-------------+--------+--------------+---------+-------+-------------+--------+--------------------------------+-------------------------+----------+---------------+----------------+-------+-------------------+--------------+
|                  id                  | birthdate  | deathdate | prefix |   first    |     last    | suffix |    maiden    | marital |  race |  ethnicity  | gender |           birthplace           |         address         |   city   |     state     |     county     |  zip  |        lat        |     lon      |
+--------------------------------------+------------+-----------+--------+------------+-------------+--------+--------------+---------+-------+-------------+--------+--------------------------------+-------------------------+----------+---------------+----------------+-------+-------------------+--------------+
| 002bc307-2fff-04ba-161b-98cce123e226 | 1933-08-13 |    None   |  Mrs.  |  Lavonia8  |  Wunsch504  |  None  |   Will178    |    M    | white | nonhispanic |   F    | Monson Center Massachusetts US |   1068 Quitzon Station  |  Quincy  | Massachusetts | Norfolk County | 02171 | 42.28861911574163 | -70.95876453 |
| 0034fe01-207f-275f-6b4b-821f7b0af044 | 1984-01-27 |    None   |  Mrs.  | Sharice315 | Johnston597 |  None  |   Yundt842   |    M    | asian | nonhispanic |   F    |   Brookline Massachusetts US   | 641 Okuneva Dale Apt 41 |  Quincy  | Massachusetts | Norfolk County | 02186 | 42.31640709165807 | -70.9685896  |
| 00abe029-00fa-a666-34f5-258a36978f6d | 1962-08-08 |    None   |  Mrs.  | Lourdes258 |  Torres807  |  None  | Santacruz647 |    M    | white |   hispanic  |   F    |     Caguas Puerto Rico PR      |  1083 Kris View Unit 89 | Weymouth | Massachusetts | Norfolk County |       | 42.24410317671621 | -70.98687798 |
+--------------------------------------+------------+-----------+--------+------------+-------------+--------+--------------+---------+-------+-------------+--------+--------------------------------+-------------------------+----------+---------------+----------------+-------+-------------------+--------------+

In [7]:
%%sql
-- Encounters table overview
SELECT *
FROM encounters
LIMIT 3;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

3 rows affected.

+--------------------------------------+---------------------+---------------------+--------------------------------------+--------------------------------------+--------------------------------------+----------------+-----------+------------------------------------------------------------------+---------------------+------------------+----------------+------------+-------------------+
|                  id                  |        start        |         stop        |               patient                |             organization             |                payer                 | encounterclass |    code   |                           description                            | base_encounter_COST | total_claim_cost | payer_coverage | reasoncode | reasondescription |
+--------------------------------------+---------------------+---------------------+--------------------------------------+--------------------------------------+--------------------------------------+----------------+-----------+------------------------------------------------------------------+---------------------+------------------+----------------+------------+-------------------+
| 0002c38a-54e9-0788-930a-90900dce3612 | 2011-08-07 09:46:50 | 2011-08-07 10:01:50 | 5e129b38-3d63-3692-2a53-c2b56f9d8af0 | d78e84ec-30aa-3bba-a33a-f29a3a454662 | 6e2f1a2d-27bd-3701-8d08-dae202c58632 |   ambulatory   | 185349003 |                      Encounter for check-up                      |        85.55        |     10224.23     |    6936.92     |    None    |        None       |
| 00059b24-6473-ca4a-8795-7373d4ddc7e0 | 2021-07-15 08:58:01 | 2021-07-15 09:13:01 | 45b87421-394d-6654-4d81-6cb5b165e455 | d78e84ec-30aa-3bba-a33a-f29a3a454662 | 6e2f1a2d-27bd-3701-8d08-dae202c58632 |   outpatient   |  33879002 | Administration of vaccine to produce active immunity (procedure) |        142.58       |      278.58      |     152.69     |    None    |        None       |
| 00091c5b-f3a1-ee7b-88cc-850c746f8f58 | 2013-07-29 01:06:15 | 2013-07-29 01:21:15 | 1712d26d-822d-1e3a-2267-0a9dba31d7c8 | d78e84ec-30aa-3bba-a33a-f29a3a454662 | b1c428d6-4f07-31e0-90f0-68ffa6ff8c76 |   outpatient   | 185349003 |                Encounter for check up (procedure)                |        85.55        |      85.55       |      0.00      |    None    |        None       |
+--------------------------------------+---------------------+---------------------+--------------------------------------+--------------------------------------+--------------------------------------+----------------+-----------+------------------------------------------------------------------+---------------------+------------------+----------------+------------+-------------------+

In [8]:
%%sql
-- Procedures table overview
SELECT *
FROM procedures
LIMIT 3;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

3 rows affected.

+---------------------+---------------------+--------------------------------------+--------------------------------------+-----------+---------------------------------------------------------+-----------+------------+-------------------+
|        start        |         STOP        |               patient                |              encounter               |    code   |                       description                       | base_cost | reasoncode | reasondescription |
+---------------------+---------------------+--------------------------------------+--------------------------------------+-----------+---------------------------------------------------------+-----------+------------+-------------------+
| 2011-01-02 09:26:36 | 2011-01-02 12:58:36 | 3de74169-7f67-9304-91d4-757e0f3a14d2 | 32c84703-2481-49cd-d571-3899d5820253 | 265764009 |                Renal dialysis (procedure)               |    903    |    None    |        None       |
| 2011-01-03 05:44:39 | 2011-01-03 06:01:42 | d9ec2e44-32e9-9148-179a-1653348cc4e2 | c98059da-320a-c0a6-fced-c8815f3e3f39 |  76601001 |                 Intramuscular injection                 |    2477   |    None    |        None       |
| 2011-01-04 14:49:55 | 2011-01-04 15:04:55 | d856d6e6-4c98-e7a2-129b-44076c63d008 | 2cfd4ddd-ad13-fe1e-528b-15051cea2ec3 | 703423002 | Combined chemotherapy and radiation therapy (procedure) |   11620   |    None    |        None       |
+---------------------+---------------------+--------------------------------------+--------------------------------------+-----------+---------------------------------------------------------+-----------+------------+-------------------+

In [9]:
%%sql
-- Payers table overview
SELECT *
FROM payers
LIMIT 3;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

3 rows affected.

+--------------------------------------+--------------+----------------------+--------------+---------------------+-------+----------------+
|                  id                  |     name     |       address        |     city     | state_headquartered |  zip  |     phone      |
+--------------------------------------+--------------+----------------------+--------------+---------------------+-------+----------------+
| 047f6ec3-6215-35eb-9608-f9dda363a44c | Cigna Health | 900 Cottage Grove Rd |  Bloomfield  |          CT         | 06002 | 1-800-997-1654 |
| 42c4fca7-f8a9-3cd1-982a-dd9751bf3e2a |    Anthem    |   220 Virginia Ave   | Indianapolis |          IN         | 46204 | 1-800-331-1476 |
| 4d71f845-a6a9-3c39-b242-14d25ef86a8d |    Aetna     |  151 Farmington Ave  |   Hartford   |          CT         | 06156 | 1-800-872-3862 |
+--------------------------------------+--------------+----------------------+--------------+---------------------+-------+----------------+

<h1 align="middle">4.0 Analysis</h1>

## 4.1. Encounters Overview
### 4.1.1. Total encounters that occurred each year.

In [10]:
%%sql
SELECT YEAR(START) AS year, COUNT(Id) AS total_encounters
FROM encounters
GROUP BY YEAR(START)
ORDER BY YEAR(START);

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

12 rows affected.

+------+------------------+
| year | total_encounters |
+------+------------------+
| 2011 |       1336       |
| 2012 |       2106       |
| 2013 |       2495       |
| 2014 |       3885       |
| 2015 |       2469       |
| 2016 |       2451       |
| 2017 |       2360       |
| 2018 |       2292       |
| 2019 |       2228       |
| 2020 |       2519       |
| 2021 |       3530       |
| 2022 |       220        |
+------+------------------+

**Encounters Overview (2011–2022):**
- Hospital encounters showed a steady **rise from 2011 to 2014, peaking at 3,885 encounters**, suggesting growth in hospital services or patient demand. Between 2015 and 2019, **encounters stabilized around 2,200 - 2,400**, indicating a period of consistent operations.
- The noticeable increase in 2020 - 2021 suggests potential COVID-19-related activity or recovery from deferred care. The sharp decline in 2022 (220 encounters) likely reflects incomplete or partial data rather than an actual reduction in visits, given that the data only has January data.

### 4.1.2. Percentage of encounters per class per year 
(ambulatory, outpatient, wellness, urgent care, emergency, and inpatient)?

In [11]:
%%sql
-- Using a CTE
WITH yearly AS (
    SELECT 
        YEAR(start) AS year,
        COUNT(id) AS year_encounters
    FROM encounters
    GROUP BY YEAR(start)
)
SELECT 
    YEAR(e.start) AS year,
    e.encounterclass,
    COUNT(e.id) AS total_encounters,
    y.year_encounters,
    ROUND(COUNT(e.id)/y.year_encounters * 100) AS pct
FROM encounters e
JOIN yearly y ON y.year = YEAR(e.start)
GROUP BY YEAR(e.start), e.encounterclass, y.year_encounters
ORDER BY year DESC;
--LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

72 rows affected.

+------+----------------+------------------+-----------------+-----+
| year | encounterclass | total_encounters | year_encounters | pct |
+------+----------------+------------------+-----------------+-----+
| 2022 |   ambulatory   |       121        |       220       |  55 |
| 2022 |   emergency    |        18        |       220       |  8  |
| 2022 |   inpatient    |        4         |       220       |  2  |
| 2022 |   outpatient   |        37        |       220       |  17 |
| 2022 |   urgentcare   |        28        |       220       |  13 |
| 2022 |    wellness    |        12        |       220       |  5  |
| 2021 |   ambulatory   |       1303       |       3530      |  37 |
| 2021 |   emergency    |       221        |       3530      |  6  |
| 2021 |   inpatient    |        55        |       3530      |  2  |
| 2021 |   outpatient   |       1418       |       3530      |  40 |
| 2021 |   urgentcare   |       378        |       3530      |  11 |
| 2021 |    wellness    |       155        |       3530      |  4  |
| 2020 |   ambulatory   |       1192       |       2519      |  47 |
| 2020 |   emergency    |       234        |       2519      |  9  |
| 2020 |   inpatient    |        72        |       2519      |  3  |
| 2020 |   outpatient   |       497        |       2519      |  20 |
| 2020 |   urgentcare   |       365        |       2519      |  14 |
| 2020 |    wellness    |       159        |       2519      |  6  |
| 2019 |   ambulatory   |       846        |       2228      |  38 |
| 2019 |   emergency    |       227        |       2228      |  10 |
| 2019 |   inpatient    |       134        |       2228      |  6  |
| 2019 |   outpatient   |       456        |       2228      |  20 |
| 2019 |   urgentcare   |       397        |       2228      |  18 |
| 2019 |    wellness    |       168        |       2228      |  8  |
| 2018 |   ambulatory   |       933        |       2292      |  41 |
| 2018 |   emergency    |       247        |       2292      |  11 |
| 2018 |   inpatient    |        81        |       2292      |  4  |
| 2018 |   outpatient   |       480        |       2292      |  21 |
| 2018 |   urgentcare   |       377        |       2292      |  16 |
| 2018 |    wellness    |       174        |       2292      |  8  |
| 2017 |   ambulatory   |       987        |       2360      |  42 |
| 2017 |   emergency    |       218        |       2360      |  9  |
| 2017 |   inpatient    |       126        |       2360      |  5  |
| 2017 |   outpatient   |       475        |       2360      |  20 |
| 2017 |   urgentcare   |       385        |       2360      |  16 |
| 2017 |    wellness    |       169        |       2360      |  7  |
| 2016 |   ambulatory   |       1073       |       2451      |  44 |
| 2016 |   emergency    |       250        |       2451      |  10 |
| 2016 |   inpatient    |       124        |       2451      |  5  |
| 2016 |   outpatient   |       481        |       2451      |  20 |
| 2016 |   urgentcare   |       341        |       2451      |  14 |
| 2016 |    wellness    |       182        |       2451      |  7  |
| 2015 |   ambulatory   |       1073       |       2469      |  43 |
| 2015 |   emergency    |       228        |       2469      |  9  |
| 2015 |   inpatient    |       111        |       2469      |  4  |
| 2015 |   outpatient   |       506        |       2469      |  20 |
| 2015 |   urgentcare   |       380        |       2469      |  15 |
| 2015 |    wellness    |       171        |       2469      |  7  |
| 2014 |   ambulatory   |       2341       |       3885      |  60 |
| 2014 |   emergency    |       216        |       3885      |  6  |
| 2014 |   inpatient    |       118        |       3885      |  3  |
| 2014 |   outpatient   |       694        |       3885      |  18 |
| 2014 |   urgentcare   |       327        |       3885      |  8  |
| 2014 |    wellness    |       189        |       3885      |  5  |
| 2013 |   ambulatory   |       1106       |       2495      |  44 

**Encounter Class Trends:**
<br>From 2011 to 2022, hospital encounters show a clear shift toward ambulatory and outpatient care, reflecting a broader move to short, non-admission visits.
- `Ambulatory` visits consistently **dominated across all years**, starting at 50% in 2011, peaking at 60% in 2014, and remaining high (around 40-55%) through 2022. This suggests a strong emphasis on routine checkups, follow-ups, and non-emergency treatments.
- `Outpatient` visits ranked **second** in share, typically between 18-24% across the years, showing constant demand for specialized care without hospital stays.
- `Emergency` and `Inpatient` encounters remained **low** (mostly 4-10% and 2-6%, respectively), indicating that most visits are non-critical, a good sign of preventive and early intervention practices.
- `Urgent care` and `Wellness` visits showed **steady growth over time**, signaling increasing attention to preventive healthcare and quick-access medical services.

By 2020–2021, during the pandemic period, ambulatory and outpatient visits together made up over 75% of total encounters, likely due to COVID-19 protocols, where hospitals prioritized outpatient and essential care while minimizing long stays.

This pattern suggests effective **community-level health management**, **preventive care initiatives**, and **efficient service delivery**, but also highlights the importance of maintaining emergency and inpatient readiness for critical cases

### 4.1.3.  Percentage of encounters over 24 hours versus under 24 hours

In [12]:
%%sql
WITH timeday AS (
    SELECT
        COUNT(*) AS total,
        SUM(CASE WHEN stop >= start + INTERVAL 24 HOUR THEN 1 ELSE 0 END) AS over_24hrs,
        SUM(CASE WHEN stop < start + INTERVAL 24 HOUR THEN 1 ELSE 0 END) AS under_24hrs
    FROM encounters)
SELECT
    ROUND(over_24hrs/total * 100,2) AS pct_over_24hrs,
    ROUND(under_24hrs/total * 100,2) AS pct_under_24hrs
FROM timeday;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

1 rows affected.

+----------------+-----------------+
| pct_over_24hrs | pct_under_24hrs |
+----------------+-----------------+
|      4.13      |      95.87      |
+----------------+-----------------+

**Treatment timeline:**
<br>Only 4.13% of encounters lasted more than 24 hours, while the majority (95.87%) were completed within a day.
- This pattern highlights that the hospital **primarily handles short-term or same-day care**, such as outpatient and ambulatory visits, with only a small proportion of patients requiring extended observation or inpatient admission.
- It suggests **strong operational efficiency** and a system focused on **quick-turnaround services**, likely supported by well-coordinated diagnostic and treatment workflows.

## 4.2. Cost & Coverage Insights
### 4.2.1. Encounters with zero payer coverage, and percentage of the total.

In [13]:
%%sql
WITH zero AS (
    SELECT COUNT(*) AS total_encounters,
    	SUM(CASE WHEN PAYER_COVERAGE=0 THEN 1 ELSE 0 END) AS zero_payer
    FROM encounters)
SELECT *, 
ROUND(zero_payer/total_encounters*100,2) AS pct_zero_payer
FROM zero;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

1 rows affected.

+------------------+------------+----------------+
| total_encounters | zero_payer | pct_zero_payer |
+------------------+------------+----------------+
|      27891       |   13586    |     48.71      |
+------------------+------------+----------------+

**Payer insights:**
<br>Out of 27,891 total hospital encounters, 13,586 (**48.7%**) had no payer coverage, meaning almost half of all visits were self-paid or uninsured.
- This finding highlights a significant portion of patients **without insurance support**, which could influence both **hospital revenue stability** and **patient access to care**. It also suggests that many patients may be relying on out-of-pocket payments, potentially leading to financial strain for both patients and the facility.

**Recommendations**
- Expand Insurance **Partnerships** – Work with more insurers and public programs to improve coverage.
- Increase Enrollment **Awareness** – Educate patients on available insurance options and registration.
- **Evaluate Financial Impact** – Compare insured vs. uninsured encounters to guide pricing and policy decisions.

### 4.2.2. The top 10 most frequent procedures performed and the average base cost for each.

In [14]:
%%sql
SELECT 
	description,
	COUNT(*) AS count_procedure,
    ROUND(AVG(BASE_COST),2) AS avg_base_cost
FROM procedures
GROUP BY description
ORDER BY count_procedure DESC
LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

10 rows affected.

+------------------------------------------------------------------------------------+-----------------+---------------+
|                                    description                                     | count_procedure | avg_base_cost |
+------------------------------------------------------------------------------------+-----------------+---------------+
|               Assessment of health and social care needs (procedure)               |       4596      |     431.00    |
|                           Hospice care (regime/therapy)                            |       4098      |     431.00    |
|                          Depression screening (procedure)                          |       3614      |     431.00    |
| Depression screening using Patient Health Questionnaire Two-Item score (procedure) |       3614      |     431.00    |
|                      Assessment of substance use (procedure)                       |       2906      |     431.00    |
|                             Renal dialysis (procedure)                             |       2746      |    1004.09    |
|                   Assessment using Morse Fall Scale (procedure)                    |       2422      |     431.00    |
|                         Assessment of anxiety (procedure)                          |       2288      |     431.00    |
|                       Medication Reconciliation (procedure)                        |       2284      |     509.12    |
|                        Screening for drug abuse (procedure)                        |       1484      |     431.00    |
+------------------------------------------------------------------------------------+-----------------+---------------+

💰**Top 10 Procedures and Average Base Cost:**
<br>The most frequent hospital procedures were primarily assessments and screenings, such as:
- Health and social care needs assessment (4,596 cases)
- Hospice care (4,098)
- Depression and anxiety screenings (over 3,600 each)
<br>Most of these procedures had an average base cost of around **$431**, reflecting standardized, low-complexity services.
<br>However, renal dialysis stood out with a significantly **higher average base cost of `$1004`** indicating its resource-intensive nature.

💡**Recommendations:**<br>
- **Prioritize Preventive Assessments** – Continue investing in screening and assessment programs to catch conditions early and reduce high-cost treatments later.
- **Review High-Cost Procedures** – Analyze cost drivers for renal dialysis and similar treatments to identify efficiency opportunities.
- **Standardize Pricing Models** – Maintain clear, consistent pricing for routine procedures to ensure transparency and affordability.

### 4.2.3. The top 10 procedures with the highest average base cost and the number of times they were performed.

In [15]:
%%sql 
SELECT 
	description,
    ROUND(AVG(BASE_COST),2) AS avg_base_cost,
    COUNT(*) AS count_procedure
FROM procedures
GROUP BY description
ORDER BY avg_base_cost DESC
LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

10 rows affected.

+---------------------------------------------------------------------------------+---------------+-----------------+
|                                   description                                   | avg_base_cost | count_procedure |
+---------------------------------------------------------------------------------+---------------+-----------------+
|                             Admit to ICU (procedure)                            |   206260.40   |        5        |
|                         Coronary artery bypass grafting                         |    47085.89   |        9        |
|                         Lumpectomy of breast (procedure)                        |    29353.00   |        5        |
|                             Hemodialysis (procedure)                            |    29299.56   |        27       |
|        Insertion of biventricular implantable cardioverter defibrillator        |    27201.00   |        4        |
|                             Electrical cardioversion                            |    25903.11   |       1383      |
|                            Partial resection of colon                           |    25229.29   |        7        |
|                Fine needle aspiration biopsy of lung (procedure)                |    23141.00   |        1        |
| Percutaneous mechanical thrombectomy of portal vein using fluoroscopic guidance |    20228.04   |        57       |
|                        Percutaneous coronary intervention                       |    19728.00   |        9        |
+---------------------------------------------------------------------------------+---------------+-----------------+

**Highest-Cost Procedures:**<br>
The top 10 highest-cost procedures reflect complex and resource-intensive hospital operations.
<br>Leading the list is **ICU admission** at an average base cost of **$206,260**, despite being performed only 5 times, showing its high critical-care expense.

Other major contributors include:
- Coronary artery bypass grafting – `$47,086` (9 times)
- Lumpectomy of breast – `$29,353` (5 times)
- Hemodialysis – `$29,300` (27 times)
- Electrical cardioversion – `$25,903` (1,383 times), notable for its frequency despite high cost per case
<br>These trend emphasizes the significant cost variation between life-saving procedures and routine screenings identified earlier.

### 4.2.4. The average total claim cost for encounters, broken down by payer.

In [16]:
%%sql
SELECT 
	p.name AS payer_name,
    ROUND(AVG(e.TOTAL_CLAIM_COST),2) AS avg_total_claim_cost
FROM encounters AS e
	LEFT JOIN payers AS p
    ON e.PAYER = p.Id
GROUP BY p.name
ORDER BY avg_total_claim_cost DESC;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

10 rows affected.

+------------------------+----------------------+
|       payer_name       | avg_total_claim_cost |
+------------------------+----------------------+
|        Medicaid        |       6205.22        |
|      NO_INSURANCE      |       5593.20        |
|         Anthem         |       4236.81        |
|         Humana         |       3269.30        |
| Blue Cross Blue Shield |       3245.58        |
|      Cigna Health      |       2996.95        |
|    UnitedHealthcare    |       2848.34        |
|         Aetna          |       2767.05        |
|        Medicare        |       2167.55        |
|     Dual Eligible      |       1696.19        |
+------------------------+----------------------+

🧾**Average Claim Cost by Payer:**
<br>The average total claim cost varied significantly across payer types, highlighting differences in coverage levels and care complexity.
- **Medicaid** recorded the highest average claim cost (`$6,205`), suggesting it supports patients with higher care needs or longer stays.
- **Uninsured** patients followed closely at `$5,593`, indicating a heavy financial burden on self-paying patients.
- **Private insurers** such as Anthem (`$4,237`) and Humana (`$3,269`) fell in the mid-range, while Medicare (`$2,168`) and Dual Eligible (`$1,696`) had the lowest average claim costs, possibly due to standardized reimbursement structures or less complex care needs.

## 4.3. Patient Behavior Analysis
### 4.3.1. Unique patients admitted each quarter over time

In [17]:
%%sql
SELECT 
    YEAR(start) AS year,
    COUNT(DISTINCT CASE WHEN QUARTER(start) = 1 THEN patient END) AS Q1,
    COUNT(DISTINCT CASE WHEN QUARTER(start) = 2 THEN patient END) AS Q2,
    COUNT(DISTINCT CASE WHEN QUARTER(start) = 3 THEN patient END) AS Q3,
    COUNT(DISTINCT CASE WHEN QUARTER(start) = 4 THEN patient END) AS Q4,
	COUNT(DISTINCT patient) AS unique_year_total
FROM encounters
--WHERE ENCOUNTERCLASS = 'inpatient'
GROUP BY YEAR(start)
ORDER BY year;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

12 rows affected.

+------+-----+-----+-----+-----+-------------------+
| year |  Q1 |  Q2 |  Q3 |  Q4 | unique_year_total |
+------+-----+-----+-----+-----+-------------------+
| 2011 | 156 | 162 | 155 | 168 |        410        |
| 2012 | 249 | 261 | 232 | 240 |        559        |
| 2013 | 243 | 271 | 259 | 242 |        570        |
| 2014 | 394 | 315 | 272 | 260 |        630        |
| 2015 | 244 | 259 | 252 | 244 |        553        |
| 2016 | 237 | 245 | 241 | 258 |        552        |
| 2017 | 235 | 256 | 234 | 243 |        546        |
| 2018 | 250 | 256 | 228 | 241 |        535        |
| 2019 | 229 | 231 | 226 | 232 |        514        |
| 2020 | 229 | 235 | 249 | 258 |        519        |
| 2021 | 417 | 414 | 279 | 223 |        649        |
| 2022 | 103 |  0  |  0  |  0  |        103        |
+------+-----+-----+-----+-----+-------------------+

🧍‍♀️**Unique Patient Admissions per Quarter:** <br>
From 2011 to 2022, the number of unique patients admitted each year showed **steady growth** in the early years, rising from **410** in 2011 to a peak of **649** in 2021.
- **2011 - 2014:** Admissions **increased consistently**, reaching 630 unique patients in 2014, suggesting expanding hospital capacity or growing community reliance on hospital care.
- **2015 - 2020:** Annual admissions **stabilized between 500–550**, reflecting a period of steady operations and consistent patient demand.
- **2021:** A sharp rise to 649 patients marks the highest annual total, potentially linked to post-pandemic healthcare rebounds or deferred treatments being resumed.
- **2022:** Only 103 patients (Q1 only) were recorded, indicating incomplete data rather than a true drop in admissions.

Quarterly patterns remained **fairly balanced** across most years, though Q1 and Q2 often showed slightly higher admissions, possibly reflecting new-year checkups or early-year health program cycles.

**Recommendations:**
- Leverage 2021 Recovery Insights – The post-pandemic increase may highlight effective patient re-engagement strategies worth maintaining.
- Monitor Quarterly Flows – Since admissions stay relatively balanced, uniform staffing and resource distribution across quarters can ensure operational stability.

### In each quarter, how many patients came in only that quarter (and never appeared in any other quarter of that year)

In [18]:
%%sql 
WITH patient_quarters AS ( -- Identify patient-quarter combination
    SELECT 
        patient,
        YEAR(start) AS yr,
        QUARTER(start) AS qtr
    FROM encounters
    GROUP BY patient, YEAR(start), QUARTER(start)
),
	patient_year_counts AS ( -- Count how many quarters each patient appears in per year
    SELECT 
        patient,
        yr,
        COUNT(DISTINCT qtr) AS quarters_in_year
    FROM patient_quarters
    GROUP BY patient, yr
),
	exclusive_patients AS ( -- Keep only patients who appear in exactly 1 quarter per year
    SELECT pq.patient, pq.yr, pq.qtr
    FROM patient_quarters pq
    JOIN patient_year_counts pyc
      ON pq.patient = pyc.patient AND pq.yr = pyc.yr
    WHERE pyc.quarters_in_year = 1
)
SELECT -- Count them per quarter/year
    yr AS Year,
    SUM(CASE WHEN qtr = 1 THEN 1 ELSE 0 END) AS Q1_Exclusive,
    SUM(CASE WHEN qtr = 2 THEN 1 ELSE 0 END) AS Q2_Exclusive,
    SUM(CASE WHEN qtr = 3 THEN 1 ELSE 0 END) AS Q3_Exclusive,
    SUM(CASE WHEN qtr = 4 THEN 1 ELSE 0 END) AS Q4_Exclusive
FROM exclusive_patients
GROUP BY yr
ORDER BY yr;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

12 rows affected.

+------+--------------+--------------+--------------+--------------+
| Year | Q1_Exclusive | Q2_Exclusive | Q3_Exclusive | Q4_Exclusive |
+------+--------------+--------------+--------------+--------------+
| 2011 |      70      |      70      |      64      |      72      |
| 2012 |      93      |      72      |      74      |      74      |
| 2013 |      90      |      89      |      75      |      68      |
| 2014 |     122      |      52      |      58      |      59      |
| 2015 |      81      |      88      |      73      |      67      |
| 2016 |      82      |      72      |      68      |      78      |
| 2017 |      80      |      90      |      76      |      68      |
| 2018 |      77      |      79      |      61      |      74      |
| 2019 |      73      |      70      |      65      |      82      |
| 2020 |      62      |      65      |      56      |      84      |
| 2021 |      93      |      77      |      38      |      23      |
| 2022 |     103      |      0       |      0       |      0       |
+------+--------------+--------------+--------------+--------------+

### 4.3.2. Number of patients readmitted within 30 days of a previous encounter.
*definition:* A **readmission** occurs when a patient is admitted (START) within 30 days of discharge (STOP) from a previous admission.

In [19]:
%%sql
WITH ordered_encounters AS (
    SELECT 
        patient,
        start,
        stop,
        LAG(stop) OVER (PARTITION BY patient ORDER BY start) AS prev_stop
    FROM encounters
)
SELECT 
    COUNT(DISTINCT patient) AS unique_patients_readmitted_30d,
    COUNT(*) AS total_readmissions_30d
FROM ordered_encounters
WHERE prev_stop IS NOT NULL
  AND TIMESTAMPDIFF(DAY, prev_stop, start) < 30;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

1 rows affected.

+--------------------------------+------------------------+
| unique_patients_readmitted_30d | total_readmissions_30d |
+--------------------------------+------------------------+
|              773               |         17312          |
+--------------------------------+------------------------+

🔁 **30-Day Readmissions:**
- A total of 773 unique patients were readmitted within 30 days of a previous encounter, resulting in 17,312 readmission events overall.

### Identifying the name and id of the patients readmited in 30 days

In [20]:
%%sql
WITH ordered_encounters AS (
    SELECT 
        patient AS patient_id,
        description,
        e.start,
        e.stop,
        LAG(stop) OVER (PARTITION BY patient ORDER BY start) AS prev_stop
    FROM encounters AS e
    -- WHERE ENCOUNTERCLASS = 'inpatient'
)
SELECT 
    oe.patient_id,
    CONCAT(pt.prefix, " ", pt.`first`," ", pt.`last`) AS patient_full_name,
    oe.prev_stop AS previous_discharge,
    oe.start AS readmission_start,
    TIMESTAMPDIFF(DAY, oe.prev_stop, oe.start) AS days_since_discharge,
    pr.description
FROM ordered_encounters AS oe
LEFT JOIN patients AS pt
    ON oe.patient_id = pt.id
LEFT JOIN procedures as pr
    ON oe.patient_id = pr.patient
WHERE prev_stop IS NOT NULL
  AND TIMESTAMPDIFF(DAY, prev_stop, oe.start) < 30
ORDER BY days_since_discharge DESC
LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

10 rows affected.

+--------------------------------------+--------------------------+---------------------+---------------------+----------------------+------------------------------------------------------------------------------------+
|              patient_id              |    patient_full_name     |  previous_discharge |  readmission_start  | days_since_discharge |                                    description                                     |
+--------------------------------------+--------------------------+---------------------+---------------------+----------------------+------------------------------------------------------------------------------------+
| 09ed0d53-f114-873c-302a-8b3b16b91de9 | Mr. Shad704 Lueilwitz711 | 2015-09-16 23:00:15 | 2015-10-16 22:45:15 |          29          |                          Depression screening (procedure)                          |
| 09ed0d53-f114-873c-302a-8b3b16b91de9 | Mr. Shad704 Lueilwitz711 | 2015-09-16 23:00:15 | 2015-10-16 22:45:15 |          29          | Depression screening using Patient Health Questionnaire Two-Item score (procedure) |
| 09ed0d53-f114-873c-302a-8b3b16b91de9 | Mr. Shad704 Lueilwitz711 | 2015-10-16 23:00:15 | 2015-11-15 22:45:15 |          29          |                          Depression screening (procedure)                          |
| 09ed0d53-f114-873c-302a-8b3b16b91de9 | Mr. Shad704 Lueilwitz711 | 2015-09-16 23:00:15 | 2015-10-16 22:45:15 |          29          |               Assessment of health and social care needs (procedure)               |
| 03751d2d-391e-a681-dbf6-4255e9f01e18 |  Mr. Les282 Stroman228   | 2020-03-01 09:17:50 | 2020-03-30 12:12:15 |          29          |                                   Clavicle X-ray                                   |
| 03751d2d-391e-a681-dbf6-4255e9f01e18 |  Mr. Les282 Stroman228   | 2020-03-01 09:17:50 | 2020-03-30 12:12:15 |          29          |                           Bone density scan (procedure)                            |
| 09ed0d53-f114-873c-302a-8b3b16b91de9 | Mr. Shad704 Lueilwitz711 | 2015-09-16 23:00:15 | 2015-10-16 22:45:15 |          29          |                       Medication Reconciliation (procedure)                        |
| 09ed0d53-f114-873c-302a-8b3b16b91de9 | Mr. Shad704 Lueilwitz711 | 2015-10-16 23:00:15 | 2015-11-15 22:45:15 |          29          | Depression screening using Patient Health Questionnaire Two-Item score (procedure) |
| 03751d2d-391e-a681-dbf6-4255e9f01e18 |  Mr. Les282 Stroman228   | 2020-03-01 09:17:50 | 2020-03-30 12:12:15 |          29          |                         Admission to orthopedic department                         |
| 09ed0d53-f114-873c-302a-8b3b16b91de9 | Mr. Shad704 Lueilwitz711 | 2015-10-16 23:00:15 | 2015-11-15 22:45:15 |          29          |               Assessment of health and social care needs (procedure)               |
+--------------------------------------+--------------------------+---------------------+---------------------+----------------------+------------------------------------------------------------------------------------+

**Patients Readmitted Within 30 Days:**<br>
Analysis revealed that a few patients experienced multiple readmissions exactly 29 days after their previous discharge, suggesting possible chronic conditions, repeated monitoring procedures, or insufficient post-discharge interventions. Key patterns identified include:
- *Mr. Shad Lueilwitz* → numerous 29-day readmissions between 2015 - 2017, primarily involving depression screening, medication reconciliation, and assessment of health and social care needs, indicating ongoing mental health and care coordination challenges.
- *Mr. Les Stroman* → multiple short-interval readmissions in 2020, mostly tied to orthopedic procedures such as clavicle X-ray and bone density scans, suggesting follow-up care for physical injuries or recovery monitoring.
- *Mr. Olin Kerluke* → repeated 29-day readmissions from 2014 - 2017, associated with health assessments and routine checkups, reflecting recurring medical supervision.
- *Mr. Raleigh Frami* → several short-term readmissions spanning 2014 - 2019, implying ongoing treatment management and possible continuity-of-care gaps.
- *Mr. Maurice Hermiston* → multiple readmissions within two years, potentially due to chronic or unmanaged health complications requiring consistent follow-up.
<br>These patterns underscore the need for stronger discharge planning, closer outpatient follow-up, and predictive readmission tracking to improve patient recovery and reduce short-term hospital returns.
<br>**Recommendation:** Establish a **readmission risk tracking system**, **implement personalized discharge follow-ups**, and integrate predictive analytics to identify and support patients likely to be readmitted within 30 days.

### 4.3.3. The patients with the most readmissions.

In [21]:
%%sql
WITH readmissions AS (
    SELECT 
        patient,
        COUNT(*) - 1 AS num_readmissions
    FROM encounters
    GROUP BY patient
),
patient_procedures AS (
    SELECT 
        e.patient AS patient_id,
        CONCAT(p.prefix, " ", p.first, " ", p.last) AS patient_full_name,
        pr.description
    FROM encounters e
    LEFT JOIN patients p 
        ON e.patient = p.id
    LEFT JOIN procedures pr 
        ON e.id = pr.encounter
)
SELECT 
    pp.patient_id,
    pp.patient_full_name,
    r.num_readmissions,
    GROUP_CONCAT(DISTINCT pp.description ORDER BY pp.description SEPARATOR ', ') AS procedures
FROM patient_procedures pp
LEFT JOIN readmissions r
    ON pp.patient_id = r.patient
GROUP BY pp.patient_id, pp.patient_full_name, r.num_readmissions
ORDER BY r.num_readmissions DESC
LIMIT 5;

Running query in 'mysql+pymysql://root:***@localhost:3306/hospital_db'

5 rows affected.

+--------------------------------------+-----------------------------+------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|              patient_id              |      patient_full_name      | num_readmissions |                                                                                                                                                                                                                                                                                                                                                                                                      procedures                                                                                                                                                                                                                                                                                                                                                                                                      |
+--------------------------------------+-----------------------------+------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1712d26d-822d-1e3a-2267-0a9dba31d7c8 | Mrs. Kimberly627 Collier206 |       1380       | Assessment of anxiety (procedure), Assessment of health and social care needs (procedure), Assessment of substance use (procedure), Assessment using Alcohol Use Disorders Identification Test - Consumption (procedure), Assessment using Morse Fall Scale (procedure), Catheter ablation of tissue of heart, Depression screening (procedure), Depression screening using Patient Health Questionnaire Nine Item score (procedure), Depression screening using Patient Health Questionnaire Two-Item score (procedure), Electrical cardioversion, Hospice care (regime/therapy), Measurement of respiratory function (procedure), Medication Reconciliation (procedure), Patient discharge (procedure), Renal dialysis (procedure), Screening for domestic abuse (procedure), Screening for drug abuse (procedure) |
| 5e055638-0dad-dfd5-005d-1e74b6fd29ac |   Mrs. Shani239 Parisian75  |       886        |         Assessment of anxiety (procedure), Assessment of health and social care needs (procedure), Assessment of substance use (procedure), Assessment using Alcohol Use Disorders Identification Test - Consumption (procedure), Assessment using Morse Fall Scale (procedure), Bone density scan (procedure), Bone immobilization, Depression screening (procedure), Depression screening using 

**Analysis of Top 5 Patients with the Most Readmissions:**<br>
Analysis revealed that several patients experienced exceptionally high numbers of readmissions, suggesting ongoing medical management, chronic health conditions, or recurring follow-up protocols. Key patterns identified include:
- **Mrs. Kimberly627 Collier206** → 1,380 readmissions, the highest among all patients. Her encounters are heavily associated with depression screening, anxiety assessments, and social and health care evaluations, indicating sustained management of mental health and psychosocial conditions, likely within a chronic care framework.
- **Mrs. Shani239 Parisian75** → 886 readmissions, primarily linked to mental health assessments, bone density scans, and renal dialysis. The combination of psychological evaluations and renal-related procedures suggests co-management of chronic physical and mental health conditions.
- **Mr. Mariano761 OKon634** → 876 readmissions, frequently associated with cardiac procedures such as coronary artery bypass grafting and echocardiography, alongside regular depression screenings. This implies long-term cardiovascular treatment coupled with psychological monitoring.
- **Mr. Gail741 Glover433** → 446 readmissions involving heart procedures (e.g., catheter ablation, electrical cardioversion), anxiety assessments, and drug abuse screenings. His case points to recurring cardiovascular issues with concurrent behavioral or substance-use assessments.
- **Mr. Ward668 Nicolas769** → 440 readmissions featuring cardiovascular stress testing, electrocardiographic procedures, and depression screenings. The pattern reflects continuous follow-up for chronic heart disease and related psychological evaluations.

**Overall Insight**<br>
The top readmitted patients share common patterns frequent psychological assessments and cardiovascular procedures, suggesting a link between chronic conditions (like heart disease) and mental health management. This trend emphasizes the importance of integrated care and post-discharge follow-up programs to reduce recurrent admissions.

<h1 align="middle"> 5.0. Recommendations </h1> 

The analysis of hospital encounter data highlights recurring readmissions, repeated short-interval encounters, and patterns of overlapping medical and psychological procedures. These insights reveal critical opportunities to improve continuity of care, optimize discharge planning, and reduce preventable readmissions.

- **Strengthen Post-Discharge Follow-Up:**<br>
Many readmissions occurred within 30 days of a previous discharge, suggesting potential gaps in post-discharge monitoring. Implementing structured follow-up calls, scheduled outpatient reviews, and targeted care coordination for high-risk patients can significantly lower short-interval readmissions.
- **Enhance Chronic Disease Management Programs:**<br>
The top readmitted patients exhibited consistent involvement in cardiovascular and mental health procedures, implying that chronic conditions drive repeated encounters. Introducing comprehensive chronic care management pathways, including patient education, medication adherence support, and regular multidisciplinary reviews, could stabilize these patients over time.
- **Integrate Mental and Physical Health Services:**<br>
Frequent depression and anxiety assessments across multiple encounters highlights the strong link between mental health and hospital utilization. Incorporating psychological support and behavioral therapy into routine medical care can improve long-term outcomes and patient resilience.
- **Adopt Predictive Monitoring and Early Intervention:**<br>
With consistent historical readmission trends, predictive analytics can be applied to identify patients at high risk of returning within 30 days. Early warning models can trigger alerts for proactive interventions, preventing repeated hospitalizations.

<h1 align="middle">6.0 Conclusion</h1>

Overall, the findings demonstrate that targeted interventions, especially those focusing on chronic condition management, post-discharge continuity, and mental health integration, are crucial for reducing readmission rates and improving patient outcomes. By combining data-driven insights with coordinated care strategies, healthcare providers can enhance service quality, reduce costs, and promote sustained patient well-being